src/api/app/views/webui/shared/bs_requests/_requests_filter.html.haml

```diff
@@ -1,95 +1,96 @@
-                                                             checked: selected_filter[:involvement] == 'incoming' }
-    = render partial: 'webui/shared/radio_button', locals: { label: 'Outgoing',
-                                                             key: 'involvement[outgoing]', name: 'involvement', value: 'outgoing',
-                                                             checked: selected_filter[:involvement] == 'outgoing' }
-.mt-4.mb-4
-  %h6.px-3.py-2
-    %b State
-  .px-3
-    .dropdown.form-multi-select{ 'data-name': 'state of request' }
-      %button.form-select.text-start#filter-state-requests-button{ type: 'button', data: { 'bs-toggle': 'dropdown' } }
-        Select the state of request
-      .dropdown-menu.w-100
-        - BsRequest::VALID_REQUEST_STATES.each do |state|
-          .dropdown-item-text
-            = render partial: 'webui/shared/check_box', locals: { label: render(BsRequestStateBadgeComponent.new(state: state)),
-                                                                  key: "state[#{state}]", name: 'state[]',
-                                                                  value: state,
-                                                                  checked: selected_filter[:state]&.include?(state.to_s) }
+.accordion.accordion-flush
+  .mt-2.mb-2.accordion-item.border-0
+    %h6.px-3.py-2.accordion-button.no-style{ data: { 'bs-toggle': 'collapse', 'bs-target': '#request-filter-involvement' },
+                                             aria: { expanded: 'true', controls: 'request-filter-involvement' } }
+      %b Involvement
+    .px-4.pb-2.accordion-collapse.collapse.show#request-filter-involvement
+      = render partial: 'webui/shared/radio_button', locals: { label: 'All',
+                                                              key: 'involvement[all]', name: 'involvement', value: 'all',
+                                                              checked: selected_filter[:involvement] == 'all' }
+      = render partial: 'webui/shared/radio_button', locals: { label: 'Incoming',
+                                                              key: 'involvement[incoming]', name: 'involvement', value: 'incoming',
+                                                              checked: selected_filter[:involvement] == 'incoming' }
+      = render partial: 'webui/shared/radio_button', locals: { label: 'Outgoing',
+                                                              key: 'involvement[outgoing]', name: 'involvement', value: 'outgoing',
+                                                              checked: selected_filter[:involvement] == 'outgoing' }
+
+  .mt-2.mb-2.accordion-item.border-0
+    %h6.px-3.py-2.accordion-button.no-style{ data: { 'bs-toggle': 'collapse', 'bs-target': '#request-filter-state' },
+                                                       aria: { expanded: 'true', controls: 'request-filter-state' } }
+      %b State
```

Comment:
We could display the current state of the filter in the header once collapsed, to avoid confusion

Reply:
What about making better use (AKA refactor/redesign) of [this](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/views/webui/shared/bs_requests/_form.html.haml#L7) instead? (which is also broken at the moment :sweat_smile: ) It would include all the selected filter values for completeness.
Maybe rendering it on the top of the right column, before the `search` bar, you know? - just my 2c
I think your solution works fine when one or two items are selected, but when there are many, I'm afraid it starts to feel crowded and a bit unorganized.

Reply:
We could also just display that something there is selected by using bold headers

Reply:
Maybe a number of selected, or the first selected option and a text that says `+3` if four options are selected

Reply:
![Peek 2024-12-12 09-59](https://github.com/user-attachments/assets/4603804a-7737-431d-ac73-2baeb60030c1)

Reply:
My only concern now is keeping this one line, but I would consider this good enough for now

Reply:
I had the same concern, but after trying it out, even if it goes on multiple lines, it does not hurt so much (it is not perfectly designed, but still it delivers the value and it works just fine). We can make it one line only by adding some css trick or something like that eventually. Let's finalize the behavior for now.